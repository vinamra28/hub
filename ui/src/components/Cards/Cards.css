.hub-resource-card {
  background-color: white;
  height: 23em;
  width: 18.5em;
  color: black;
  border-radius: 0.5em;
}

.hub-rating-icon {
  color: #484848;
  height: 1.7em;
  width: 1.7em;
}

.hub-resource-name {
  font-weight: normal;
  font-size: 1.3em;
}

.hub-resource-version {
  float: right;
  font-weight: normal;
  margin-top: 0.3em;
}

.hub-resource-updatedAt {
  margin-bottom: 0.5em;
  margin-top: -0.3em;
}

.hub-tags {
  margin-right: 0.3em;
  margin-bottom: 0.5em;
}

.hub-kind-icon {
  margin-right: 0.5em;
  padding-top: 0.125em;
}

.hub-catalog-icon {
  vertical-align: -0.125em;
}

.hub-card-link {
  text-decoration: none;
}

.div {
  color: black;
}

.hub-resource-body {
  overflow: hidden;
  text-overflow: ellipsis;
  margin-bottom: 1em;
  max-height: 10em;
}

.fade {
  position: relative;
}
.fade:after {
  content: '';
  text-align: right;
  position: absolute;
  bottom: 0;
  right: 0;
  width: 70%;
  height: 1.6em;
  background: linear-gradient(to right, rgba(255, 255, 255, 0), rgba(255, 255, 255, 1) 50%);
}

.hub-spinner {
  position: fixed;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  margin: auto;
}
