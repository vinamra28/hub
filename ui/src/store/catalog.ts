import { types, Instance } from 'mobx-state-tree';

export const CatalogType = types
  .model({
    id: types.integer,
    name: types.string,
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type ICatalogType = Instance<typeof CatalogType>;
export type ICatalogStore = Instance<typeof CatalogStore>;

export const CatalogStore = types
  .model('CatalogStore', {
    list: types.array(CatalogType)
  })

  .views((self) => ({
    get count(): number {
      return self.list.length;
    },
    get catalogs(): Array<string> {
      const { list } = self;
      return list
        .filter((type: ICatalogType) => type.selected)
        .map((type: ICatalogType) => type.name);
    }
  }))

  .actions((self) => ({
    add(item: ICatalogType) {
      self.list.push(item);
    },
    clear() {
      self.list.map((item: ICatalogType) => (item.selected = false));
    }
  }));
