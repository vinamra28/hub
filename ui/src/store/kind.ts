import { types, Instance } from 'mobx-state-tree';

export const Kind = types
  .model('Kind', {
    id: types.integer,
    name: types.string,
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type IKind = Instance<typeof Kind>;
export type IKindStore = Instance<typeof KindStore>;

export const KindStore = types
  .model('KindStore', {
    list: types.array(Kind)
  })

  .views((self) => ({
    get count(): number {
      return self.list.length;
    },
    get kinds(): Array<string> {
      const { list } = self;
      return list.filter((type: IKind) => type.selected).map((type: IKind) => type.name);
    }
  }))

  .actions((self) => ({
    add(item: IKind) {
      self.list.push(item);
    },
    clear() {
      self.list.map((item: IKind) => (item.selected = false));
    }
  }));
