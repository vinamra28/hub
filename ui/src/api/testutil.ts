import { Api } from './';
import * as fs from 'fs';
import { ICategory } from '../store/category';
import { IResource } from '../store/resources';

export class FakeHub implements Api {
  dataDir: string;

  constructor(dataDir: string) {
    this.dataDir = dataDir;
  }

  async resources() {
    const data = `${this.dataDir}/resources.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IResource>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async categories() {
    const data = `${this.dataDir}/categories.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<ICategory>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }
}
