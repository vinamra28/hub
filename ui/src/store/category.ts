import { types, getEnv, flow, Instance } from 'mobx-state-tree';
import { Api } from '../api';

export const Tag = types.model('Tags', {
  id: types.integer,
  name: types.string
});

export const Category = types
  .model('Category', {
    id: types.number,
    name: types.string,
    tags: types.array(Tag),
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type ICategory = Instance<typeof Category>;
export type ITag = Instance<typeof Tag>;

export const CategoryStore = types
  .model('CategoryStore', {
    items: types.array(Category),
    isLoading: true,
    err: ''
  })

  .views((self) => ({
    get api(): Api {
      return getEnv(self).api;
    },

    get count() {
      return self.items.length;
    },

    get tags() {
      return self.items
        .filter((c) => c.selected)
        .reduce((acc: string[], c: ICategory) => [...acc, ...c.tags.map((t) => t.name)], []);
    }
  }))

  .actions((self) => ({
    add(item: ICategory) {
      self.items.push(item);
    },

    setLoading(l: boolean) {
      self.isLoading = l;
    },

    clear() {
      self.items.map((c: ICategory) => (c.selected = false));
    }
  }))

  .actions((self) => ({
    load: flow(function* () {
      try {
        self.setLoading(true);
        const { api } = self;
        const json = yield api.categories();
        json.data.forEach((item: ICategory) => self.add(item));
      } catch (err) {
        self.err = err.toString();
      }
      self.setLoading(false);
    })
  }))

  .actions((self) => ({
    afterCreate() {
      self.load();
    }
  }));

export type ICategoryStore = Instance<typeof CategoryStore>;
