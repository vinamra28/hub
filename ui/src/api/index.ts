import axios from 'axios';
import { API_URL } from '../config/constants';
import { ICategory } from '../store/category';
import { IResource } from '../store/resource';

export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
}

export class Hub implements Api {
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resources() {
    try {
      return axios.get(`${API_URL}/resources`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resourceVersion(resourceId: string) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/versions`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }
}
