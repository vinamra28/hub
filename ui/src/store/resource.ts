import { types, Instance } from 'mobx-state-tree';
import fuzzysort from 'fuzzysort';
import moment, { Moment } from 'moment';
import { flow, getEnv } from 'mobx-state-tree';
import { Tag, ICategoryStore, ITag } from './category';
import { Api } from '../api';
import { Catalog, CatalogStore } from './catalog';
import { Kind, KindStore } from './kind';

export const updatedAt = types.custom<string, Moment>({
  name: 'momentDate',
  fromSnapshot(value: string): Moment {
    return moment(new Date(value));
  },
  toSnapshot(value: Moment): string {
    return value.fromNow();
  },
  isTargetType(v: string) {
    return moment.isMoment(v);
  },
  getValidationMessage(v: string) {
    if (moment.isMoment(v)) {
      return 'Invalid moment object';
    }
    return '';
  }
});

const Version = types.model('Version', {
  id: types.identifierNumber,
  version: types.string,
  displayName: types.string,
  description: types.string,
  minPipelinesVersion: types.string,
  rawURL: types.string,
  webURL: types.string,
  updatedAt: updatedAt
});

export const Resource = types.model('Resource', {
  id: types.number,
  name: types.identifier,
  catalog: types.reference(Catalog),
  kind: types.reference(Kind),
  latestVersion: types.reference(Version),
  tags: types.array(types.reference(Tag)), // ["1", "2"]
  rating: types.number,
  versions: types.array(types.reference(Version)),
  displayName: ''
});

export type IResource = Instance<typeof Resource>;
export type IVersion = Instance<typeof Version>;

export enum sortByFields {
  Name = 'Name',
  Rating = 'Rating'
}

export const ResourceStore = types
  .model('ResourceStore', {
    resources: types.map(Resource),
    versions: types.map(Version),
    catalogs: types.optional(CatalogStore, {}),
    kinds: types.optional(KindStore, {}),
    sortBy: types.optional(types.enumeration(Object.values(sortByFields)), sortByFields.Name),
    tags: types.optional(types.map(Tag), {}),
    search: '',
    err: '',
    isLoading: true
  })
  .views((self) => ({
    get items() {
      return Array.from(self.resources.values());
    },
    get api(): Api {
      return getEnv(self).api;
    },
    get categories(): ICategoryStore {
      return getEnv(self).categories;
    }
  }))

  .actions((self) => ({
    setLoading(l: boolean) {
      self.isLoading = l;
    },
    setSearch(text: string) {
      self.search = text;
    },
    add(item: IResource) {
      self.resources.put(item);
    },
    setSortBy(field: string) {
      const key: sortByFields = sortByFields[field as keyof typeof sortByFields];
      self.sortBy = key;
    }
  }))

  .actions((self) => ({
    load: flow(function* () {
      try {
        self.setLoading(true);

        const { api } = self;
        const json = yield api.resources();

        const kinds: string[] = json.data.map((r: IResource) => r.kind);
        kinds.forEach((k) => self.kinds.add(k));

        json.data.forEach((r: IResource) => {
          self.catalogs.add(r.catalog);
          self.versions.put(r.latestVersion);
        });

        // adding the tags to the store - normalized
        const tags: ITag[] = json.data.flatMap((item: IResource) => item.tags);

        tags.forEach((t) => (t != null ? self.tags.put(t) : null));

        const resources: IResource[] = json.data.map((r: IResource) => ({
          id: r.id,
          name: r.name,
          catalog: r.catalog.id,
          kind: r.kind,
          latestVersion: r.latestVersion.id,
          tags: r.tags != null ? r.tags.map((tag: ITag) => tag.id) : [],
          rating: r.rating,
          versions: [],
          displayName: r.latestVersion.displayName
        }));

        resources.forEach((r: IResource) => {
          r.versions.push(r.latestVersion);
          self.add(r);
        });
      } catch (err) {
        self.err = err.toString();
      }
      self.setLoading(false);
    })
  }))

  .actions((self) => ({
    afterCreate() {
      self.load();
    }
  }))

  .views((self) => ({
    get filteredResources() {
      const { resources, kinds, catalogs, categories, search } = self;
      const { selectedTags } = categories;

      const filtered: IResource[] = [];
      resources.forEach((r: IResource) => {
        const matchesKind = kinds.selected.size === 0 || kinds.selected.has(r.kind.name);
        const matchesCatalogs = catalogs.selected.size === 0 || catalogs.selected.has(r.catalog.id);
        const matchesTags = selectedTags.size === 0 || r.tags.some((t) => selectedTags.has(t.id));

        if (matchesKind && matchesCatalogs && matchesTags) {
          filtered.push(r);
        }
      });

      if (search.trim() !== '') {
        return fuzzysort
          .go(search, filtered, { keys: ['name', 'displayName'] })
          .map((resource: Fuzzysort.KeysResult<IResource>) => resource.obj);
      }
      return filtered;
    }
  }));

export type IResourceStore = Instance<typeof ResourceStore>;
