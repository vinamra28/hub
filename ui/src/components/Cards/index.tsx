import React from 'react';
import {
  Card,
  CardHeader,
  CardActions,
  CardTitle,
  CardBody,
  TextContent,
  Badge,
  CardFooter,
  GalleryItem,
  Spinner
} from '@patternfly/react-core';
import { StarIcon } from '@patternfly/react-icons';
import './Cards.css';
import { IResource } from '../../store/resource';
import { ITag } from '../../store/category';
import { Link } from 'react-router-dom';
import Icon from '../Icon';
import { IconSize } from '@patternfly/react-icons';

export const resourceName = (name: string, displayName: string) => {
  return displayName === '' ? <span>{name}</span> : <span>{displayName}</span>;
};

interface Props {
  items: IResource[];
}

const Cards: React.FC<Props> = (resources) => {
  if (resources.items.length === 0) {
    return <Spinner className="hub-spinner" />;
  }

  return (
    <React.Fragment>
      {resources.items.map((resource: IResource, r: number) => (
        <GalleryItem key={r}>
          <Link
            to={{
              pathname: `${resources.items[r].catalog.name.toLowerCase()}/${resources.items[
                r
              ].kind.name.toLowerCase()}/${resources.items[r].name.toLowerCase()}/${
                resources.items[r].latestVersion.version
              }`
            }}
            className="hub-card-link"
          >
            <Card className="hub-resource-card">
              <CardHeader>
                <span className="hub-kind-icon">
                  <Icon
                    id={resources.items[r].kind.icon}
                    size={IconSize.sm}
                    label={resources.items[r].kind.name}
                  />
                </span>

                <span className="hub-catalog-icon">
                  <Icon
                    id={resources.items[r].catalog.icon}
                    size={IconSize.sm}
                    label={resources.items[r].catalog.name}
                  />
                </span>

                <CardActions>
                  <StarIcon />
                  <TextContent className="text">{resources.items[r].rating}</TextContent>
                </CardActions>
              </CardHeader>

              <CardTitle>
                <span className="hub-resource-name">
                  {resourceName(
                    resources.items[r].name,
                    resources.items[r].latestVersion.displayName
                  )}
                </span>
                <span className="hub-resource-version">
                  v{resources.items[r].latestVersion.version}
                </span>
              </CardTitle>

              <CardBody className="hub-resource-body fade">
                {resources.items[r].latestVersion.description}
              </CardBody>

              <CardFooter>
                <TextContent className="hub-resource-updatedAt">
                  Updated {resources.items[r].latestVersion.updatedAt.fromNow()}
                </TextContent>

                <div style={{ height: '2.5em' }}>
                  {resources.items[r].tags.slice(0, 3).map((tag: ITag) => (
                    <Badge className="hub-tags" key={`badge-${tag.id}`}>
                      {tag.name}
                    </Badge>
                  ))}
                </div>
              </CardFooter>
            </Card>
          </Link>
        </GalleryItem>
      ))}
    </React.Fragment>
  );
};

export default Cards;
