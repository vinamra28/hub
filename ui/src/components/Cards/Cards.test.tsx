import React from 'react';
import { when } from 'mobx';
import { shallow } from 'enzyme';
import { FakeHub } from '../../api/testutil';
import { ResourceStore } from '../../store/resource';
import { CategoryStore } from '../../store/category';
import Cards, { resourceName } from '.';

const TESTDATA_DIR = `src/store/testdata`;
const api = new FakeHub(TESTDATA_DIR);

describe('Resources', () => {
  it('should render the resources', (done) => {
    const store = ResourceStore.create({}, { api, categories: CategoryStore.create({}, { api }) });

    when(
      () => !store.isLoading,
      () => {
        const component = shallow(<Cards items={store.filteredResources} />);
        expect(component.debug()).toMatchSnapshot();

        done();
      }
    );
  });

  it('should display the name', () => {
    const wrapper = shallow(resourceName('tekton', ''));
    expect(wrapper.find('span').text()).toBe('tekton');
  });
});
