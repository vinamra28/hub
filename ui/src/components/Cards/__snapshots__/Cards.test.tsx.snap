// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resources should render the resources 1`] = `
"<Fragment>
  <GalleryItem>
    <Link to={{...}} className=\\"hub-card-link\\">
      <Card className=\\"hub-resource-card\\">
        <CardHeader>
          <span className=\\"hub-kind-icon\\">
            <Icon id=\\"build\\" size=\\"sm\\" label=\\"Task\\" />
          </span>
          <span className=\\"hub-catalog-icon\\">
            <Icon id=\\"user\\" size=\\"sm\\" label=\\"tekton\\" />
          </span>
          <CardActions>
            <StarIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false} />
            <TextContent className=\\"text\\">
              5
            </TextContent>
          </CardActions>
        </CardHeader>
        <CardTitle>
          <span className=\\"hub-resource-name\\">
            <span>
              aws cli
            </span>
          </span>
          <span className=\\"hub-resource-version\\">
            v
            0.1
          </span>
        </CardTitle>
        <CardBody className=\\"hub-resource-body fade\\">
          This task performs operations on Amazon Web Services resources using aws.
        </CardBody>
        <CardFooter>
          <TextContent className=\\"hub-resource-updatedAt\\">
            Updated 
            a month ago
          </TextContent>
          <div style={{...}}>
            <Badge className=\\"hub-tags\\">
              cli
            </Badge>
          </div>
        </CardFooter>
      </Card>
    </Link>
  </GalleryItem>
  <GalleryItem>
    <Link to={{...}} className=\\"hub-card-link\\">
      <Card className=\\"hub-resource-card\\">
        <CardHeader>
          <span className=\\"hub-kind-icon\\">
            <Icon id=\\"build\\" size=\\"sm\\" label=\\"Task\\" />
          </span>
          <span className=\\"hub-catalog-icon\\">
            <Icon id=\\"user\\" size=\\"sm\\" label=\\"tekton\\" />
          </span>
          <CardActions>
            <StarIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false} />
            <TextContent className=\\"text\\">
              5
            </TextContent>
          </CardActions>
        </CardHeader>
        <CardTitle>
          <span className=\\"hub-resource-name\\">
            <span>
              golang build
            </span>
          </span>
          <span className=\\"hub-resource-version\\">
            v
            0.1
          </span>
        </CardTitle>
        <CardBody className=\\"hub-resource-body fade\\">
          This Task is Golang task to build Go projects.
        </CardBody>
        <CardFooter>
          <TextContent className=\\"hub-resource-updatedAt\\">
            Updated 
            a month ago
          </TextContent>
          <div style={{...}}>
            <Badge className=\\"hub-tags\\">
              build-tool
            </Badge>
          </div>
        </CardFooter>
      </Card>
    </Link>
  </GalleryItem>
  <GalleryItem>
    <Link to={{...}} className=\\"hub-card-link\\">
      <Card className=\\"hub-resource-card\\">
        <CardHeader>
          <span className=\\"hub-kind-icon\\">
            <Icon id=\\"build\\" size=\\"sm\\" label=\\"Task\\" />
          </span>
          <span className=\\"hub-catalog-icon\\">
            <Icon id=\\"user\\" size=\\"sm\\" label=\\"tekton\\" />
          </span>
          <CardActions>
            <StarIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false} />
            <TextContent className=\\"text\\">
              5
            </TextContent>
          </CardActions>
        </CardHeader>
        <CardTitle>
          <span className=\\"hub-resource-name\\">
            <span>
              jenkins operation
            </span>
          </span>
          <span className=\\"hub-resource-version\\">
            v
            0.1
          </span>
        </CardTitle>
        <CardBody className=\\"hub-resource-body fade\\">
          The following task can be used to interact with the the Jenkins REST API.
        </CardBody>
        <CardFooter>
          <TextContent className=\\"hub-resource-updatedAt\\">
            Updated 
            a month ago
          </TextContent>
          <div style={{...}}>
            <Badge className=\\"hub-tags\\">
              build
            </Badge>
            <Badge className=\\"hub-tags\\">
              jenkins
            </Badge>
          </div>
        </CardFooter>
      </Card>
    </Link>
  </GalleryItem>
  <GalleryItem>
    <Link to={{...}} className=\\"hub-card-link\\">
      <Card className=\\"hub-resource-card\\">
        <CardHeader>
          <span className=\\"hub-kind-icon\\">
            <Icon id=\\"build\\" size=\\"sm\\" label=\\"Task\\" />
          </span>
          <span className=\\"hub-catalog-icon\\">
            <Icon id=\\"user\\" size=\\"sm\\" label=\\"tekton\\" />
          </span>
          <CardActions>
            <StarIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false} />
            <TextContent className=\\"text\\">
              5
            </TextContent>
          </CardActions>
        </CardHeader>
        <CardTitle>
          <span className=\\"hub-resource-name\\">
            <span>
              jib maven
            </span>
          </span>
          <span className=\\"hub-resource-version\\">
            v
            0.1
          </span>
        </CardTitle>
        <CardBody className=\\"hub-resource-body fade\\">
          This Task builds Java/Kotlin/Groovy/Scala source into a container image using Googleâ€™s Jib tool.
          Jib works with Maven and Gradle projects, and this template is for Maven projects.
        </CardBody>
        <CardFooter>
          <TextContent className=\\"hub-resource-updatedAt\\">
            Updated 
            a month ago
          </TextContent>
          <div style={{...}}>
            <Badge className=\\"hub-tags\\">
              image-build
            </Badge>
          </div>
        </CardFooter>
      </Card>
    </Link>
  </GalleryItem>
  <GalleryItem>
    <Link to={{...}} className=\\"hub-card-link\\">
      <Card className=\\"hub-resource-card\\">
        <CardHeader>
          <span className=\\"hub-kind-icon\\">
            <Icon id=\\"build\\" size=\\"sm\\" label=\\"Task\\" />
          </span>
          <span className=\\"hub-catalog-icon\\">
            <Icon id=\\"user\\" size=\\"sm\\" label=\\"tekton\\" />
          </span>
          <CardActions>
            <StarIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false} />
            <TextContent className=\\"text\\">
              4.5
            </TextContent>
          </CardActions>
        </CardHeader>
        <CardTitle>
          <span className=\\"hub-resource-name\\">
            <span>
              Hub Pipeline
            </span>
          </span>
          <span className=\\"hub-resource-version\\">
            v
            0.1
          </span>
        </CardTitle>
        <CardBody className=\\"hub-resource-body fade\\">
          Hub is a platform which displays all resources
        </CardBody>
        <CardFooter>
          <TextContent className=\\"hub-resource-updatedAt\\">
            Updated 
            a month ago
          </TextContent>
          <div style={{...}}>
            <Badge className=\\"hub-tags\\">
              cli
            </Badge>
          </div>
        </CardFooter>
      </Card>
    </Link>
  </GalleryItem>
  <GalleryItem>
    <Link to={{...}} className=\\"hub-card-link\\">
      <Card className=\\"hub-resource-card\\">
        <CardHeader>
          <span className=\\"hub-kind-icon\\">
            <Icon id=\\"domain\\" size=\\"sm\\" label=\\"Pipeline\\" />
          </span>
          <span className=\\"hub-catalog-icon\\">
            <Icon id=\\"cat\\" size=\\"sm\\" label=\\"tekton-hub\\" />
          </span>
          <CardActions>
            <StarIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false} />
            <TextContent className=\\"text\\">
              4.5
            </TextContent>
          </CardActions>
        </CardHeader>
        <CardTitle>
          <span className=\\"hub-resource-name\\">
            <span>
              Hub Pipeline
            </span>
          </span>
          <span className=\\"hub-resource-version\\">
            v
            0.1
          </span>
        </CardTitle>
        <CardBody className=\\"hub-resource-body fade\\">
          Hub is a platform which displays all resources
        </CardBody>
        <CardFooter>
          <TextContent className=\\"hub-resource-updatedAt\\">
            Updated 
            a month ago
          </TextContent>
          <div style={{...}}>
            <Badge className=\\"hub-tags\\">
              resource
            </Badge>
          </div>
        </CardFooter>
      </Card>
    </Link>
  </GalleryItem>
</Fragment>"
`;
